pkg/acceptance/helpers/common.go

```diff
@@ -19,3 +24,23 @@ func EnsureQuotedIdentifiersIgnoreCaseIsSetToFalse(client *sdk.Client, ctx conte
 	}
 	return nil
 }
+
+// MatchAllStringsInOrder returns a regex matching every string in parts
+func MatchAllStringsInOrder(parts []string) *regexp.Regexp {
+	escapedParts := make([]string, len(parts))
+	for i := range parts {
+		escapedParts[i] = regexp.QuoteMeta(parts[i])
+	}
+	return regexp.MustCompile(strings.Join(escapedParts, "((.|\n)*)"))
+}
+
+// AssertErrorContainsPartsFunc returns a function asserting error message contains each string in parts
+func AssertErrorContainsPartsFunc(t *testing.T, parts []string) resource.ErrorCheckFunc {
```

Comment:
so why do we need a second method doing almost the same thing as the one above?

And also: this one should also be tested, but because assertions are a bit trickier to test (but we have examples in our repository), it does not have to be a part of this PR.

Reply:
Because for some errors, e.g. missing required fields in config, the order of errors is not deterministic. We get error message as a simple `error` so the alternative is to parse it and sort messages and I'm not a fan of such implementation. Also, function signatures are different because they are used as `resource.TestStep.ExpectError` and `resource.TestCase.ErrorCheck` respectively.

I can add tests in the next PR.

Reply:
Okay , but why:
- do we need `ErrorCheck` versus `ExpectError`? Why can't we just use expect error too here?
- do we care about the ordering of errors at all then? Maybe one assertion not caring about the ordering would be enough?